hls-graph/src/Development/IDE/Graph/Internal/Types.hs

```diff
@@ -202,11 +203,10 @@ data RunResult value = RunResult
         -- ^ The value to store in the Shake database.
     ,runValue   :: value
         -- ^ The value to return from 'Development.Shake.Rule.apply'.
+    ,runHook    :: STM ()
+        -- ^ The hook to run after the rule completes.
```

Comment:
Maybe give an example of what this is for? e.g. "This is useful if the rule needs to modify state which is being tracked outside the rule system"?

Reply:
Although I guess it's then not clear to me why this has to happen in this hook rather than in the action. I'm sure you know - write it down somewhere!

Reply:
done